### PR TITLE
Add !giveaway Command

### PR DESCRIPTION
**Pull request by QuirkySquid**
Giveaways in streams can be difficult when people are afk/don't want the item/are bots, so this command would streamline it to run the giveaway just once. 

USAGE:
!giveaway start [time] (Mod-Only) ----- Starts the giveaway countdown for a certain time, defaults to 60
!giveaway join -------------------------- Join the giveaway while it is counting down
!giveaway end (Mod-Only) ------------ Stops the countdown and picks a winner.
!giveaway cancel (Mod-Only) --------- Cancels the giveaway without picking a winner.

There are probably some ways to make it work more efficiently, as I am a skrub and I'm not that good at programming, but it works.

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -335,3 +334,75 @@ class SubCommand(Command):
...
+
+    def __call__(self, args, data):
+        if len(args) == 1:
+            return "Not enough arguements!"
+        if args[1].lower() == "join":
+            if self.is_running:
+                if data["user_name"] in self.candidates:
+                    return "You have already entered!"
+                else:
+                    print(self.candidates)
```
Comment (Innectic) [inline models.py]: Remove this print.

**Comment by Innectic** [comment]
Once you fix that, I'll get it merged into the next release :smile:

**Comment by Innectic** [comment]
Alright. I'll get this into the next release.

**Comment by 2Cubed** [comment]
Silly Inn. ;P

**Comment by Innectic** [comment]
Actually closing since v0.3 isn't maintained anymore. This will be in v0.4. 🥞

**Pull request closed without merging**